packages/smooth_app/lib/widgets/smooth_product_carousel.dart

```diff
@@ -279,9 +279,7 @@ class _SearchCardTagLine extends StatelessWidget {
               future: _fetchData(),
               builder: (BuildContext context,
                   AsyncSnapshot<Map<String, dynamic>> data) {
-                if (data.connectionState != ConnectionState.done ||
-                    data.data == null ||
-                    !data.hasData) {
+                if (data.data == null || !data.hasData) {
```

Comment:
You probably didn't write the initial code but that is redundant as:
```dart
bool get hasData => data != null;
```

In addition to that, the `data.connectionState != ConnectionState.done` (or `==`, whatever) must be dealt with explicitly, as this is the only thing that really matters in `AsyncSnapshot`.

Reply:
> You probably didn't write the initial code but that is redundant as:
> 
> ```dart
> bool get hasData => data != null;
> ```
> 
> In addition to that, the `data.connectionState != ConnectionState.done` (or `==`, whatever) must be dealt with explicitly, as this is the only thing that really matters in `AsyncSnapshot`.

@monsieurtanuki Then we will have to handle `ConnectionState.waiting` in some other way .let say some sort of loading indicator or shimmer effect, But that too would be just replacing default tagline text with some other thing, Which will cause blinking issue.

Reply:
> Then we will have to handle ConnectionState.waiting in some other way.

@omkarChend1kar I don't understand everything but if you have to, you have to.

It's not just a matter of fixing a bug, it's a matter of making code maintainable. And the first thing I would say if I had to read the code is: "where is `done` dealt with?".
In addition to that, your specific fix doesn't make any sense in a first approach, as dealing with `AsyncSnapshot` in the standard lazy way would consider that `data.connectionState != ConnectionState.done`, `data.data == null` and `!data.hasData` are the same thing (which is correct unless the data returned is actually `null`).

Besides, when it's `done` it's `done`, it doesn't look like it can go back to `waiting` later.
```dart
/// Connected to a terminated asynchronous computation.
done,
```

My questions:
* are you able to reproduce the bug?
* does your fix actually fixes the bug?
* would you consider replacing your fix with a more reasonable `data.connectionState != ConnectionState.done` test?
* I have a hunch that the problem comes from the `AutoSizeText` (a stateFULwidget that gets refreshed) - could you try with a `Text` instead (a stateLESSwidget)?

Reply:
@monsieurtanuki I am not sure why it's emitting `connectionState.waiting` during lifecycle changes, With data being not null whereas documentation suggests otherwise [https://api.flutter.dev/flutter/widgets/ConnectionState.html](url) and this is essentially what causing the blinking issue, That's why only thing, I could think was to remove the flag.
![Waiting state](https://user-images.githubusercontent.com/71365710/206753635-59d238a2-f418-47ea-8ac7-810183fcc84f.png)

Reply:
Then you're not lucky:
> [waiting](https://api.flutter.dev/flutter/widgets/widgets/ConnectionState.html), indicating that the asynchronous operation has begun, typically with the data being null.

Again, I would suggest not to rely on `data.hasData` (especially as it's unexpectedly populated while still `waiting`) but _only_ on `data.connectionState != ConnectionState.done`.
You want to display data when it's complete, don't you?

Reply:
> > Then we will have to handle ConnectionState.waiting in some other way.
> 
> @omkarChend1kar I don't understand everything but if you have to, you have to.
> 
> It's not just a matter of fixing a bug, it's a matter of making code maintainable. And the first thing I would say if I had to read the code is: "where is `done` dealt with?". In addition to that, your specific fix doesn't make any sense in a first approach, as dealing with `AsyncSnapshot` in the standard lazy way would consider that `data.connectionState != ConnectionState.done`, `data.data == null` and `!data.hasData` are the same thing (which is correct unless the data returned is actually `null`).
> 
> Besides, when it's `done` it's `done`, it doesn't look like it can go back to `waiting` later.

I probably should have added mockup in template itself, But if you check the mockup added by @teolemon at #3417 it says, When he resumes the app, Which is probably in between the lifecycle changes of the app, As per my debugging analysis after the `connectionState.done` it again emits `connectionState.waiting` that's when  flag of `data.connectionState.done!=ConnectionState.done` is true and due to this it shows default text and again state jumps back to `ConnectionState.done` and we have the blinking issue, I am yet not sure why it's happening. 
> 
> ```dart
> /// Connected to a terminated asynchronous computation.
> done,
> ```
> 
> My questions:
> 
> * are you able to reproduce the bug?

Yes!!
> * does your fix actually fixes the bug?

Yes!!
> * would you consider replacing your fix with a more reasonable `data.connectionState != ConnectionState.done` test?
> * I have a hunch that the problem comes from the `AutoSizeText` (a stateFULwidget that gets refreshed) - could you try with a `Text` instead (a stateLESSwidget)?

Reply:
> Then you're not lucky:
> 
> > [waiting](https://api.flutter.dev/flutter/widgets/widgets/ConnectionState.html), indicating that the asynchronous operation has begun, typically with the data being null.
> 
> Again, I would suggest not to rely on `data.hasData` (especially as it's unexpectedly populated while still `waiting`) but _only_ on `data.connectionState != ConnectionState.done`. You want to display data when it's complete, don't you?

@monsieurtanuki Yeah, This is what ideally should be done.

Reply:
@omkarChend1kar Part of the problem is that you're displaying that within a `Consumer` that gets refreshed.
Could you add a `print` statement at the beginning of method `_fetchData`?
I suspect this method gets called several times, and that would cause the blinking.
If it's not that, you need to investigate "in the literature" in which conditions `done` can be followed by `waiting`.

Reply:
> @omkarChend1kar Part of the problem is that you're displaying that within a `Consumer` that gets refreshed. Could you add a `print` statement at the beginning of method `_fetchData`? I suspect this method gets called several times, and that would cause the blinking. 

Yeah, It's quite possible, Will check, Thank You @monsieurtanuki!!

>If it's not that, you need to investigate "in the literature" in which conditions `done` can be followed by `waiting`.

Reply:
If so, a solution would be to wait until `done`, to store the data in a private field and to display it.
If refreshed, we should still display the private field until the next `done` where the private field value would be replaced.
That said, I also suggest that you use a different language (if you don't already), as part of the PR is about "then goes back to English".

Reply:
@monsieurtanuki Yes indeed ,You are correct, Consumer is rebuilding entire `FutureBuilder`.